clash-lib/src/Clash/Backend/SystemVerilog.hs

```diff
@@ -492,6 +501,22 @@ funDec ty@(RTree n elTy) | typeSize elTy > 0 = Just A.<$>
     ranges = brackets (int 0 <> colon <> int (2^n-1)) <>
              brackets (int (typeSize elTy - 1) <> colon <> int 0)
 
+funDec ty@(MemBlob n m) | typeSize ty > 0 = Just A.<$>
+  "function" <+> "automatic" <+> "logic" <+> ranges <+> tName <> "_to_lv" <> parens (sigDecl "i" ty) <> semi <> line <>
+  indent 2
+    ("for" <+> parens ("int n = 0" <> semi <+> "n <" <+> int n <> semi <+> "n=n+1") <> line <>
+      indent 2 (tName <> "_to_lv" <> brackets "n" <+> "=" <+> "i[n]" <> semi)) <> line <>
+  "endfunction" <> line <>
+  "function" <+> "automatic" <+> tName <+> tName <> "_from_lv" <> parens ("logic" <+> ranges <+> "i") <> semi <> line <>
+  indent 2
+    ("for" <+> parens ("int n = 0" <> semi <+> "n <" <+> int n <> semi <+> "n=n+1") <> line <>
+      indent 2 (tName <> "_from_lv" <> brackets "n" <+> "=" <+> "i[n]" <> semi)) <> line <>
+  "endfunction" <> line
+  where
+    tName  = tyName ty
+    ranges = brackets (int 0 <> colon <> int (n-1)) <>
+             brackets (int (m - 1) <> colon <> int 0)
+
```

Comment:
I don't think we want to convert `MemBlob`s to logic vectors in BlackBox templates. We only ever want to use them as literals for memory-like functions. So this can be deleted.

Instead, you probably want to explicitly error out on MemBlob in the to/from SLV conversions: https://github.com/clash-lang/clash-compiler/blob/d087788ded8721c60a912b05015bdb28e5d36aaa/clash-lib/src/Clash/Backend/SystemVerilog.hs#L1268-L1290

Reply:
I don't know, why not give users the option? People are creative; they might think of a completely new purpose for `MemBlob` in their fancy primitive, one where they want to look at the data differently with `toSLV`. And even though the actual `MemBlob` is only ever a `~CONST`, people might still conceivably want some `fromSLV` to make non-constant data the same shape as the `MemBlob`. They could write the relevant SystemVerilog conversion functions themselves, but we could also provide it to them here.

Reply:
```haskell
toSLV t e = case t of
  MemBlob n m -> toSLV (Vector n (BitVector m)) e

fromSV (MemBlob n m) = fromSLV (Vector n (BitVector m))

simpleFromSLV (MemBlob n m) = simpleFromSLV (Vector n (BitVector m))

expFromSLV (MemBlob n m) = expFromSLV (Vector n (BitVector m))
```

Reply:
Yeah, that works for me